### ISSUE TITLE
Mac OSX & Ubuntu 16.04: ERROR:  permission denied to create extension "unaccent"

### ISSUE THREAD

**Issue by Vincent14** [sleede/fab-manager#36]
In [this](https://github.com/LaCasemate/fab-manager#general-guidelines) documentation section, at step 8:

> $ rake db:setup
> fablab_development already exists
> -- enable_extension("plpgsql")
>    -> 0.0568s
> -- enable_extension("unaccent")
> rake aborted!
> ActiveRecord::StatementInvalid: PG::InsufficientPrivilege: ERROR:  permission denied to create extension "unaccent"
> HINT:  Must be superuser to create this extension.
> : CREATE EXTENSION IF NOT EXISTS "unaccent"

But I get the same error even if I do it:

> postgres=# CREATE EXTENSION IF NOT EXISTS "unaccent";
> CREATE EXTENSION
> postgres=#

Do you have any idea?

**Comment by sylvainbx** [comment]
As the error message said:

> ERROR: permission denied to create extension "unaccent"
> HINT: Must be superuser to create this extension.

I guess the user you configured in `database.yml` was not granted full permissions on the fab-manager's database. Maybe you can try something like the following, in `psql`:

``` sql
GRANT ALL PRIVILEGES ON fabmanager_development TO myuser;
```

_Edit:_ which version of PostgreSQL do you use?

**Comment by Vincent14** [comment]
Le mercredi 17 août 2016 22:43:23 CEST, Sylvain notifications@github.com 
a écrit :

> As the error message said:
> 
> > ERROR: permission denied to create extension "unaccent"
> > HINT: Must be superuser to create this extension.
> 
> I guess the user you configured in `database.yml` was not 
> granted full permissions on the fab-manager's database. Maybe 
> you can try something like the following, in `psql`:
> 
> ```
>     GRANT ALL PRIVILEGES ON fabmanager_development TO myuser;
> ```

Well, I've just followed the doc from two different clean installs, should 
I use the 'sleede' user for that? Isn't already done by step 5 (CREATE 
DATABASE fablab_development OWNER sleede)?

## 

Message envoyé via Dekko depuis mon appareil Ubuntu

**Comment by sylvainbx** [comment]
Yes it is. Sounds strange that it doesn't work for you. 
Can you specify which version of PostgreSQL do you use ?

**Comment by Vincent14** [comment]
I am using v9.5 on Ubuntu, and the version given by brew on Mac (pretty sure it's the 9.4).

\l :

> ```
>                                     List of databases
>       Name          |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges   
> ```
> 
> ------------------------+----------+----------+-------------+-------------+-----------------------
>  fablab_development     | sleede   | UTF8     | fr_FR.UTF-8 | fr_FR.UTF-8 | 
>  fablab_test            | sleede   | UTF8     | fr_FR.UTF-8 | fr_FR.UTF-8 | 
>  fabmanager_development | sleede   | UTF8     | fr_FR.UTF-8 | fr_FR.UTF-8 | 
>  fabmanager_test        | sleede   | UTF8     | fr_FR.UTF-8 | fr_FR.UTF-8 | 
>  postgres               | postgres | UTF8     | fr_FR.UTF-8 | fr_FR.UTF-8 | 
>  template0              | postgres | UTF8     | fr_FR.UTF-8 | fr_FR.UTF-8 | =c/postgres          +
>                         |          |          |             |             | postgres=CTc/postgres
>  template1              | postgres | UTF8     | fr_FR.UTF-8 | fr_FR.UTF-8 | =c/postgres          +
>                         |          |          |             |             | postgres=CTc/postgres
> (7 rows)

**Comment by sylvainbx** [comment]
Okey, I've looked around this issue today and I was able to reproduce it with (Ubuntu 16.04/OS X 10.11) + PostgreSQL 9.5.

I'm not 100% sure, but this issue may be related to the version of Postgre as we've always used PG 9.4 in the past and never experienced this issue. I need to investigate further on this point to confirm.

In the meantime, you can workaround this issue in a development environment using the following command in `psql`. This is needed anyway for running the test suite (see [known issues](https://github.com/LaCasemate/fab-manager#known-issues)):

``` sql
ALTER USER sleede WITH SUPERUSER;
```

In any case, for a production environment I would definitely recommend that you [use docker](https://github.com/LaCasemate/fab-manager/blob/master/docker/README.md).

**Comment by Vincent14** [comment]
It works! Thank you very much! You should edit the doc :)

**Comment by sylvainbx** [comment]
I re-open this issue as a reference, because we will update the docs.

**Comment by sylvainbx** [comment]
This was fixed in commit referenced above on the [dev](https://github.com/LaCasemate/fab-manager/tree/dev) branch. These changes will be released with next stable version (probably 2.4.0).
